### ISSUE TITLE
Incorrect docstring in irc.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3148]
|<img alt="Wolf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Wolf reported|
|-|-|
|Trac ID|trac#3148|
|Type|defect|
|Created|2008-04-02 13:01:44Z|

In the irc.py module, on the line 702 ([http://twistedmatrix.com/trac/browser/trunk/twisted/words/protocols/irc.py#L702]), there's written "Called when a channel's modes are changed" but the method is called even if the modes of a user are changed:

```
class MyClient(irc.IRCClient):
    def modeChanged(self, user, channel, set, modes, args):
        print 'Mode changed:', user, channel, set, modes, args

# Output:
# just after the connection to the server
Mode changed: mynick mynick True ix ()
# when someone ops me
Mode changed: opper!~x@foobarbaz #chan True o ('mynick',)
# when someone deops someone else
Mode changed: deopper!~x@foobarbaz #chan False o ('deopped',)
# when someone sets +c on a channel
Mode changed: someone!~x@foobarbaz #chan True c ()
```

Only in the last of these 4 examples the change is relative to a channel. In the 2nd and 3rd examples, the changes are relative to a user in a channel and in the 1st to a user.

If is not possible to have a method for the user modes and another method for the channel modes, the docstring should be corrected in order to warn the user that modeChanged handles *all* the changes and not only the ones relative to the channel.

<details><summary>Searchable metadata</summary>

```
trac-id__3148 3148
type__defect defect
reporter__Wolf Wolf
priority__low low
milestone__Twisted_8_2 Twisted-8.2
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__irc_IRCClient_modeChanged irc IRCClient modeChanged
time__1207141304000000 1207141304000000
changetime__1211530854000000 1211530854000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Milestone Words-8.0.0+1 deleted

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

It has been closed in r23671 with [#3230](https://github.com/twisted/twisted/issues/3230).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
